src/native/eventpipe/ds-eventpipe-protocol.c

```diff
@@ -336,6 +370,62 @@ ds_eventpipe_collect_tracing2_command_payload_free (EventPipeCollectTracing2Comm
 	ep_rt_object_free (payload);
 }
 
+/*
+* EventPipeCollectTracing3CommandPayload
+*/
+
+static
+uint8_t *
+eventpipe_collect_tracing3_command_try_parse_payload (
```

Comment:
Ref previous comment, when we are in the process of extending this command, again, maybe its time to consider making it more extensible.

Reply:
I think my main concern with more extensible is making sure we are keeping the tool informed about what settings are supported and which ones aren't. For example lets say we added a general purpose key/value property bag to the message. I think we would still need to return a failure code if we encounter any key/value pair that isn't recognized and then the caller could either retry with fewer keys or give up because the runtime is too old to support the behavior they wanted. Effectively we'd be creating a 2nd layer of the versioning scheme where rather than falling back to earlier versions of the command they are falling back to smaller subsets of keys. I'm not sure how much improvement that would offer?

Another approach to extensibility that seems like more work in the short term but might have more useful behavior long term would be to break up the single collect command into separate create and start commands. For example a tool might issue a series of commands:
create_session
configure_session_stacktraces
configure_session_cool_new_feature
start_session

That would allow for new pieces of configuration to be added in the future without aggregating all other pre-existing config options again. It also opens the door to having configuration options that can be changed after the session is started. For example today it is impossible to switch which events are enabled without using two separate sessions and the two sessions will either overlap and duplicate events or there will be a gap where events may have been missed. If there was a configure_session_providers command that was allowed to be called while the session was running that difficulty could be avoided.

I'm not trying to force that we need to do a larger change like that as a pre-requisite for a stack-trace option, just sharing ideas at this point.